### DISCUSSION TITLE
How to 'carry' other Rigid Bodies with the Virtual Character Controller?

### DISCUSSION THREAD

**Discussion by fredUmlaut** [jrouwe/JoltPhysics#541] [General]
How could the virtual character controller pick up another Rigid Body, carry it around, and get influenced by its collision as well?
Usually you would use a joint but this would not work with the virtual character to my understanding.

Imagine picking up a crate and running around with it, but when the crate collides with something then the 
virtual character should also not be able to move on.

**Comment by jrouwe** [comment]
See #239 for info on how to create an internal rigid body. You can then attach your constraints to it, but I have to say that it's very easy to get into a feedback loop: crate gets pushed, crate pushes virtual character, inner rigid body is moved by the virtual character, crate attached to that body moves, crate collides with virtual character again, crate pushes virtual character etc. You might be better off with a kinematic crate and a ContactListener to listen for collisions with the crate and do the pushing in game code. You have to experiment, I can't tell you what will work best.
